### PR TITLE
✅ New decimals auction test

### PR DESCRIPTION
**Pull request by JuaniRios**
## What?
- Add a test for checking a wide range of decimals for CTs work as intended.

## Why?
- We might have unexpected bugs when the decimals are too high or too low

## How?
- Similar to the evaluation test `different_decimals_ct_works_as_expected`

### CODE REVIEW

**Comment by JuaniRios** [comment]
* **#275** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/275?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: nodes/parachain/src/chain_spec/polimec.rs [hunk 1]
```diff
@@ -180,7 +180,7 @@ pub fn get_rococo_chain_spec() -> GenericChainSpec {
 }
 
 fn base_testnet_genesis(
-	stakers: Vec<(AccountId, Option<AccountId>, Balance)>,
+	#[allow(unused)] stakers: Vec<(AccountId, Option<AccountId>, Balance)>,
```
Comment (vstam1) [inline nodes/parachain/src/chain_spec/polimec.rs:183]: Why is this needed?

File: nodes/parachain/src/chain_spec/politest.rs [hunk 2]
```diff
@@ -192,7 +192,7 @@ pub fn get_live_chain_spec() -> GenericChainSpec {
 
 #[allow(clippy::too_many_arguments)]
 fn testnet_genesis(
-	stakers: Vec<(AccountId, Option<AccountId>, Balance)>,
+	#[allow(unused)] stakers: Vec<(AccountId, Option<AccountId>, Balance)>,
```
Comment (vstam1) [inline nodes/parachain/src/chain_spec/politest.rs:195]: Same for this?

File: pallets/funding/src/tests/3_auction.rs [hunk 3]
```diff
@@ -516,6 +516,172 @@ mod round_flow {
...
+					.cycle();
+
+			let mut min_bid_amounts_ct = Vec::new();
+			let mut min_bid_amounts_usd = Vec::new();
+			let mut auction_allocations_ct = Vec::new();
+			let mut auction_allocations_usd = Vec::new();
+
+			let mut decimal_test = |decimals: u8| {
+				let funding_asset = funding_assets_cycle.next().unwrap();
+				dbg!(&funding_asset);
```
Comment (vstam1) [inline pallets/funding/src/tests/3_auction.rs:578]: Pls remove dbg

File: pallets/funding/src/tests/3_auction.rs [hunk 4]
```diff
@@ -516,6 +516,172 @@ mod round_flow {
...
+				let funding_asset_usd_price = match funding_asset {
+					AcceptedFundingAsset::USDT => usdt_price,
+					AcceptedFundingAsset::USDC => usdc_price,
+					AcceptedFundingAsset::DOT => dot_price,
+				};
+
+				dbg!(decimals);
```
Comment (vstam1) [inline pallets/funding/src/tests/3_auction.rs:585]: same

File: pallets/funding/src/tests/3_auction.rs [hunk 5]
```diff
@@ -516,6 +516,172 @@ mod round_flow {
...
+				project_metadata.minimum_price =
+					<TestRuntime as Config>::PriceProvider::calculate_decimals_aware_price(
+						original_price,
+						USD_DECIMALS,
+						decimals,
+					)
+					.unwrap();
+
+				dbg!(original_price);
+				dbg!(project_metadata.minimum_price);
```
Comment (vstam1) [inline pallets/funding/src/tests/3_auction.rs:597]: Same

File: pallets/funding/src/tests/3_auction.rs [hunk 6]
```diff
@@ -516,6 +516,172 @@ mod round_flow {
...
+					default_evaluators(),
+					default_weights(),
+				);
+				let project_id = inst.create_auctioning_project(project_metadata.clone(), issuer, evaluations);
+
+				let auction_allocation_percentage = project_metadata.auction_round_allocation_percentage;
+				let auction_allocation_ct = auction_allocation_percentage * project_metadata.total_allocation_size;
+				auction_allocations_ct.push(auction_allocation_ct);
+				dbg!(&project_metadata.minimum_price);
+				dbg!(&auction_allocation_ct);
```
Comment (vstam1) [inline pallets/funding/src/tests/3_auction.rs:614]: same

File: pallets/funding/src/tests/3_auction.rs [hunk 7]
```diff
@@ -516,6 +516,172 @@ mod round_flow {
...
+				inst.mint_foreign_asset_to(vec![UserToForeignAssets::new(
+					BIDDER_1,
+					min_professional_bid_funding_asset,
+					funding_asset.to_assethub_id(),
+				)]);
+
+				dbg!(&min_professional_bid_funding_asset);
+				dbg!(&min_professional_bid_usd);
+				dbg!(&min_professional_bid_ct);
+				dbg!(&min_professional_bid_plmc);
```
Comment (vstam1) [inline pallets/funding/src/tests/3_auction.rs:643]: same

File: pallets/funding/src/tests/3_auction.rs [hunk 8]
```diff
@@ -516,6 +516,172 @@ mod round_flow {
...
+					),
+					project_id,
+					min_professional_bid_ct,
+					1u8.try_into().unwrap(),
+					funding_asset,
+				)));
+
+				// The bucket should have 1MM * 10^decimals CT minus what we just bid
+				let bucket = inst.execute(|| Buckets::<TestRuntime>::get(project_id).unwrap());
+				assert_eq!(bucket.amount_left, 500_000u128 * 10u128.pow(decimals as u32) - min_professional_bid_ct);
```
Comment (vstam1) [inline pallets/funding/src/tests/3_auction.rs:661]: Comment says 1MM while value is 500_000?

**Review by vstam1 — COMMENTED** [review]
General test looks good. Please remove the dbg statements and maybe replace with asserts where possible/if it makes sense.

File: nodes/parachain/src/chain_spec/polimec.rs (hunk 1 above)
Comment (JuaniRios) [reply nodes/parachain/src/chain_spec/polimec.rs:183]: because compiling with runtime-benchmarks gives a warning

File: nodes/parachain/src/chain_spec/politest.rs (hunk 2 above)
Comment (JuaniRios) [reply nodes/parachain/src/chain_spec/politest.rs:195]: because compiling with runtime-benchmarks gives a warning

**Review by vstam1 — APPROVED** [review]

**Comment by JuaniRios** [comment]
### Merge activity

* **May 2, 5:12 AM EDT**: @JuaniRios started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Polimec/polimec-node/275).
